src/main/java/dev/j3fftw/litexpansion/machine/WaterSprinkler.java

```diff
@@ -32,7 +32,9 @@
  */
 public class WaterSprinkler extends AbstractGrowthAccelerator {
 
-    private final Set<Material> crops = EnumSet.noneOf(Material.class);
+    private final ItemSetting<Double> success_chance = new ItemSetting<>("success-chance", 0.5);
```

Comment:
You should also validate this in the constructor (and warn + re-set if needed)

Reply:
Ive tried using validateInput but I can only get it to use the closest valid double, could u explain how to set this up?

Reply:
constructor() {
  if (successChance.getValue() > 0 && successChance.getValue() < 1) {
    "Success chance has to be between 0-1, resetting to default (0.5)"
    successChance.update(0.5)
  }
}

Reply:
Can't use this inside if WaterSprinkler() because the setting hasnt been initialized